### ISSUE TITLE
Option to avoid reduction of parameter

### ISSUE THREAD

**Issue by clementval** [omni-compiler/xcodeml-tools#99]
In some cases, it would be nice to avoid reduction of parameter by the front-end. 

Imagine that you have lots of code using MPI. Therefore, you create at some point a `mpi.xmod` file that you will use for various translation. This file contains a parameter for `MPI_COMM_WORLD` with a value determined by the flavour of MPI (for example OpenMPI) you used for the `.xmod` file generation. 

When translating code using this module, you end up with the actual value of the parameter and this may not work with another flavour of MPI (mvapich). 

A nice front-end option `-no-parameter-reduction` would be nice to have.

**Comment by h-murai** [comment]
@clementval When you use F_Front as a component of a compiler, which is the original usage of F_Front we supposed, I think you must use an identical `mpif.h` or `mpif.mod` to compile all of your files. However, when using F_Front as a component of a tool or something (e.g. CLAW), such an option could be useful.

**Comment by clementval** [comment]
@clementval Exactly, it's often not necessary but it might be useful in some cases. I'll propose smth. This issue is just here to remind me of the potential option.
